apps/base-docs/tutorials/docs/2_coinbase-smart-wallet.md

```diff
@@ -0,0 +1,694 @@
+</div>
+```
+
+Add a `<header>` element to the top (or as appropriate for the UI/UX library you are using). Move the Connect `<div>` inside, and replace the `connectors.map` function with one that finds the Smart Wallet and uses it to connect. First, create the function:
+
+```typescript
+const createWallet = useCallback(() => {
+  const coinbaseWalletConnector = connectors.find(
+    (connector) => connector.id === 'coinbaseWalletSDK',
+  );
+  if (coinbaseWalletConnector) {
+    connect({ connector: coinbaseWalletConnector, chainId });
+  }
+}, [connectors, connect]);
+```
+
+Then update the `<header>` to use it. You can also move the `Disconnect` button into the header. Only show the appropriate button, depending on the connection state.
+
+```typescript
+<header>
+  <div>
+    <h2>Connect</h2>
+    {account.status === 'disconnected' && (
+      <button onClick={() => createWallet()} type="button">
+        Connect
+      </button>
+    )}
+    {account.status === 'connected' && (
+      <button type="button" onClick={() => disconnect()}>
+        Disconnect
+      </button>
+    )}
+    <div>{'Status: ' + status}</div>
+    <div>{error?.message}</div>
+  </div>
+</header>
+```
+
+### Showing the User's Balance
+
+Experienced users will know that they need funds for gas and payments to interact with your app, but new users may not. Both benefit from seeing their balance shown on the app. Import `useBalance` and initialize it below `useAccount`:
+
+```typescript
+const account = useAccount();
+const balance = useBalance({ address: account.address });
+```
+
+Then add it to the display when the user is logged in. You'll need to extract it from the object returned by `useBalance`, and you'll want to truncate some decimals. It's also a good idea to indicate the chain the user is on.
+
+Create a helper function to show the balance with four decimals:
+
+```typescript
+function weiToEtherString(wei: bigint) {
+  const ether = formatEther(wei);
+  return parseFloat(ether).toFixed(4).toString();
+}
+```
+
+Then use it to display the user's balance:
+
+```typescript
+<div>
+  {account.status === 'connected' &&
+    'Balance: (' + account.chain?.name + ') ' + weiToEtherString(balance?.data?.value || BigInt(0))}
+</div>
+```
+
+### Making it Easy to Fund Wallets
+
+One of the most confusing things about onchain apps for new users is the number of steps and the amount of knowledge it takes to get the right amount of the right currency on the right network to be able to use these apps.
+
+**The most important thing you can do to help your app and _every_ app on Base is to make it easy for _anyone_ to fund their wallet.**
+
+In doing so, you make it much easier for people to onboard to your app, and you can take advantage of being a part of a community of developers doing the same.
+
+Together, we can create a gateway to bring the world onchain!
+
+Start by logging in to [Coinbase Commerce]. Create an account if you don't have one yet.
```

Comment:
Onramp is now in Coinbase Developer Platform. I would avoid mentioning commerce to reduce product names